extend/registration/deployment.md

```diff
@@ -0,0 +1,230 @@
+
+## Manual Deployment
+If you want to use some other continuous integration setting or deploy to the repository manually, you can do so without limitations.
+As in the [above script](/extend/registration/deployment/#step-2-----add-deploy-script),
+we recommend using [Developer Portal CLI client](https://github.com/keboola/developer-portal-cli-v2). This CLI tools (runnable in Docker or PHP)
+allows you to obtain the repository for an application and push credentials to that repository. If you want to get even more low level, you
+can also use the [Developer Portal API](http://docs.kebooladeveloperportal.apiary.io/#) which has the same functions. It also allows you
+to generate [Generate credentials for a service account](http://docs.kebooladeveloperportal.apiary.io/#reference/0/vendors/generate-credentials-for-service-account)
+programmatically.
+
+## Test Live Configurations
+Before deploying your new image to the developer portal, you may want to try it out on some 'real' configurations
+in your project. You can do this by adding some environment variables to Travis with an appropriate
+[Storage token](https://help.keboola.com/storage/tokens/)
+and configuration Id. It is highly recommended to create a dedicated token for this task.
+
+The commands will need two extra environment variables apart from the
+[ones listed above](/extend/registration/deployment/#step-4-----configure-deploy-automation):
+
+- `KBC_STORAGE_TOKEN` --- the Storage token that the test(s) will run under
+- `KBC_APP_TEST_CONFIG` --- ID of the configuration to test
+
+{: .image-popup}
+![Screenshot -- Sample Configurations](/extend/registration/configuration-sample.png)
+
+If you are still using our [sample application code](https://github.com/keboola/docs-deploy-example-tests),
+use the [sample table](/extend/source.csv) as input.
+
+Then add the following steps to your `script` section in `.travis.yml` to run the test jobs.
+
+{% highlight yaml %}
+language: bash
+
+services:
+  - docker
+
+script:
+  - docker build --tag=my-application .
+  # push master image to ECR
+  - docker pull quay.io/keboola/developer-portal-cli-v2:latest
+  - export REPOSITORY=`docker run --rm -e KBC_DEVELOPERPORTAL_USERNAME -e KBC_DEVELOPERPORTAL_PASSWORD -e KBC_DEVELOPERPORTAL_URL quay.io/keboola/developer-portal-cli-v2:latest ecr:get-repository $KBC_DEVELOPERPORTAL_VENDOR $KBC_DEVELOPERPORTAL_APP`
+  - docker tag $KBC_APP_REPOSITORY:latest $REPOSITORY:master
+  - eval $(docker run --rm -e KBC_DEVELOPERPORTAL_USERNAME -e KBC_DEVELOPERPORTAL_PASSWORD -e KBC_DEVELOPERPORTAL_URL quay.io/keboola/developer-portal-cli-v2:latest ecr:get-login $KBC_DEVELOPERPORTAL_VENDOR $KBC_DEVELOPERPORTAL_APP)
+  - docker push $REPOSITORY:master
+  # Run live test job on new master image
+  - docker pull quay.io/keboola/syrup-cli:latest
+  - docker run --rm -e KBC_STORAGE_TOKEN quay.io/keboola/syrup-cli:latest run-job $KBC_DEVELOPERPORTAL_APP $KBC_APP_TEST_CONFIG master
```

Comment:
trochu se tady začaly motat `master` a `latest`, klonil bych se všechno dělat jako `latest` a nebo striktně oddělit `test`?

Reply:
Jo, teď jsem si vzpomněl (trknul mě tvůj commit v #scm), že jsme se tuším s Vladem bavili o tom, že by vlastně ani tag `test` nemusel bejt potřeba. Build/test/deploy by byl následující

Failující integrační test

 1. pushnu commit do `master`u s tagem 1.0.0
 2. zbuilduje se keboola/myapp:1.0.0
 3. keboola/myapp:1.0.0 se pushne do ECR
 4. spustí se testovací syrup job nad tagem 1.0.0
 5. test vyfailuje

Úspěšnej integrační test

 1-4. detto

 5. test proběhne úspěšně
 6. pushnu do developer portálu informaci, že appka má novej tag 1.0.0

To, že se v ECR budou stackovat verze nám nevadí (je tam limit 1000 a dá se udělat automatickej GC) a máš jistotu, že ten otestovanej image v repository je 1:1 totéž, co se pak bude pouštět. Rollback taky jde udělat snadno - developer portál ví, co bylo použito za image v předchozí verzi komponenty.

Reply:
no tak tohle uz jsme jednou zavrhli - jediny co tim ziskas, je ze se budou v ECR tagovat neotestovany image - timpadem prestane obsah toho ECR repozitare jako takovyho davat smysl - budou v nem vsechny nefunkcni veci, co jsi kdy udelal (a pro to, abys zjistil, co je funkcni musis jit nekam uplne jinam (do dev portalu)). Otestovanej image je uz dneska 1:1 to co se bude poustet

Reply:
Jasný, OK!